generated/nimodinst/tests/test_modinst.py

```diff
@@ -72,3 +70,47 @@ def test_iterating(self):
             assert len(session) == 2
             for d in session:
                 pass
+
+    def test_get_attribute_session(self):
+        val = 123
+        self.side_effects_helper['OpenInstalledDevicesSession']['deviceCount'] = 1
+        self.patched_library.niModInst_GetInstalledDeviceAttributeViInt32.side_effect = self.side_effects_helper.niModInst_GetInstalledDeviceAttributeViInt32
+        self.side_effects_helper['GetInstalledDeviceAttributeViInt32']['attributeValue'] = val
+        with nimodinst.Session('') as session:
+            attr_int = session[0].chassis_number
+            assert(attr_int == val)
+
+    def test_get_attribute_session_no_index(self):
+        val = 123
+        self.side_effects_helper['OpenInstalledDevicesSession']['deviceCount'] = 1
+        self.patched_library.niModInst_GetInstalledDeviceAttributeViInt32.side_effect = self.side_effects_helper.niModInst_GetInstalledDeviceAttributeViInt32
+        self.side_effects_helper['GetInstalledDeviceAttributeViInt32']['attributeValue'] = val
+        with nimodinst.Session('') as session:
+            try:
+                session.chassis_number
+                assert False
+            except AttributeError:
+                pass
+
+    def test_get_attribute_session_index_wrong_location(self):
+        val = 123
+        self.side_effects_helper['OpenInstalledDevicesSession']['deviceCount'] = 1
+        self.patched_library.niModInst_GetInstalledDeviceAttributeViInt32.side_effect = self.side_effects_helper.niModInst_GetInstalledDeviceAttributeViInt32
+        self.side_effects_helper['GetInstalledDeviceAttributeViInt32']['attributeValue'] = val
+        with nimodinst.Session('') as session:
+            try:
+                session.chassis_number[0]
+                assert False
+            except AttributeError:
+                pass
+
+    def test_get_attribute_for_loop(self):
+        val = 123
+        self.patched_library.niModInst_GetInstalledDeviceAttributeViInt32.side_effect = self.side_effects_helper.niModInst_GetInstalledDeviceAttributeViInt32
+        self.side_effects_helper['GetInstalledDeviceAttributeViInt32']['attributeValue'] = val
+        self.side_effects_helper['OpenInstalledDevicesSession']['deviceCount'] = 1
```

Comment:
loop with a single device is not very interested...
for instance we cannot test that the index is incremented and passed correctly.

Reply:
Add looping tests with multiple devices